src/Avalonia.Controls/Platform/ScreenHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using Avalonia.Utilities;
+
+#nullable enable
+
+namespace Avalonia.Platform
+{
+    public static class ScreenHelper
+    {
+        public static Screen? ScreenFromPoint(PixelPoint point, IReadOnlyList<Screen> screens)
+        {
+            foreach (Screen screen in screens)
+            {
+                if (screen.Bounds.Contains(point))
```

Comment:
As described [here](https://github.com/AvaloniaUI/Avalonia/issues/7164#issuecomment-994761322) `Rect.Contains` is inclusive on the right/bottom edge and so could give the wrong result here.

Reply:
I think we need to add a `Rect.ContainsExclusive` method or something, as IMO the `Rect.Contains` logic is just plain confusing (it's inherited from WPF).

Reply:
fixed